EncountifyAPI/Controllers/LocationsController.cs

```diff
@@ -0,0 +1,222 @@
+        {
+            if (name != "") EditLocationName(id, name);
+            if (description != "") EditLocationDescription(id, description);
+            if (longitude != 0) EditLocationLongitude(id, longitude);
+            if (latitude != 0) EditLocationLatitude(id, latitude);
+            if (category != 0) EditLocationCategory(id, category);
+            if (image != "") EditLocationImage(id, image);
+
+            return GetLocation(id);
+        }
+
+
+        /// <summary>
+        /// Edit an existing location's name
+        /// </summary>
+        [HttpPut("{id}/Name")]
+        public IEnumerable<Location> EditLocationName(int id, string name)
+        {
+            ExecuteQuery("UPDATE Locations SET Name = @name WHERE Id = @id", id: id, name: name);
+            return GetLocation(id);
+        }
+
+        /// <summary>
+        /// Edit an existing location's description
+        /// </summary>
+        [HttpPut("{id}/Description")]
+        public IEnumerable<Location> EditLocationDescription(int id, string description)
+        {
+            ExecuteQuery("UPDATE Locations SET Description = @description WHERE Id = @id", id: id, description: description);
+            return GetLocation(id);
+        }
+
+        /// <summary>
+        /// Edit an existing location's longitude
+        /// </summary>
+        [HttpPut("{id}/Longitude")]
+        public IEnumerable<Location> EditLocationLongitude(int id, float longitude)
+        {
+            ExecuteQuery("UPDATE Locations SET Longitude = @longitude WHERE Id = @id", id: id, longitude: longitude);
+            return GetLocation(id);
+        }
+
+        /// <summary>
+        /// Edit an existing location's latitude
+        /// </summary>
+        [HttpPut("{id}/Latitude")]
+        public IEnumerable<Location> EditLocationLatitude(int id, float latitude)
+        {
+            ExecuteQuery("UPDATE Locations SET Latitude = @latitude WHERE Id = @id", id: id, latitude: latitude);
+            return GetLocation(id);
+        }
+
+        /// <summary>
+        /// Edit an existing location's category
+        /// </summary>
+        [HttpPut("{id}/Category")]
+        public IEnumerable<Location> EditLocationCategory(int id, int category)
+        {
+            ExecuteQuery("UPDATE Locations SET Category = @category WHERE Id = @id", id: id, category: category);
+            return GetLocation(id);
+        }
+
+        /// <summary>
+        /// Edit an existing location's image
+        /// </summary>
+        [HttpPut("{id}/Image")]
+        public IEnumerable<Location> EditLocationImage(int id, string image)
+        {
+            ExecuteQuery("UPDATE Locations SET Image = @image WHERE Id = @id", id: id, image: image);
+            return GetLocation(id);
+        }
+
+        /// <summary>
+        /// Delete all locations
+        /// </summary>
+        [HttpDelete]
+        public IEnumerable<Location> DeleteAllLocations()
```

Comment:
Since this one deletes everything, I don't think, that it needs to return anything